File: 01/d01-1.f90
program d01_1
  implicit none

  integer :: fd, iostat, imaxcal, maxcal, curcal, cal
  integer, allocatable :: cals(:)
  character(100) :: buffer

  curcal = 0
  allocate(cals(0))
  open(newunit=fd, file="input.dat", form="formatted", action="read")
  do
    read(fd, "(a)", iostat=iostat) buffer
    if (iostat /= 0) exit
    if (len_trim(buffer) > 0) then
      read(buffer, *) cal
      curcal = curcal + cal
    else
      cals = [cals, curcal]
      curcal = 0
    end if
  end do
  close(fd)
  imaxcal = maxloc(cals, dim=1)
  maxcal = cals(imaxcal)
  print "(a, i0, a, i0)", "Elf with most calories: ", imaxcal, " calories: ", maxcal

end program d01_1

File: 01/d01-1.py
import numpy as np

curcal = 0
cals = []
with open("input.dat", "r") as fp:
    lines = [line.strip() for line in fp]
for line in lines:
    if line:
        curcal = curcal + int(line)
    else:
        cals.append(curcal)
        curcal = 0

imaxcal = np.argmax(cals, axis=0)
maxcal = cals[imaxcal]
print(f"Elf with most calories: {imaxcal + 1:d}, calories: {maxcal:d}")
